lib/openhab/core/dsl/items/datetime_item.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+require 'forwardable'
+require 'java'
+require 'time'
+require 'openhab/core/dsl/types/datetime'
+
+module OpenHAB
+  module Core
+    module DSL
+      module Items
+        #
+        # Delegation to OpenHAB Number Item
+        #
+        class DateTimeItem
+          extend Forwardable
+
+          def_delegator :@datetime_item, :to_s
+
+          def initialize(datetime_item)
+            @datetime_item = datetime_item
+            super()
+          end
+
+          def to_dt
+            OpenHAB::Core::DSL::Types::DateTime.new(@datetime_item.state)
+          end
+
+          def <=>(other)
+            logger.trace("Comparing self (#{self.class}) to #{other} (#{other.class})")
+            case other
+            when DateTimeItem
+              to_dt <=> other.to_dt
+            else
+              to_dt <=> other
+            end
+          end
+
+          def <(other)
```

Comment:
Does implementing comparable not work here for some reason?

Reply:
It probably does, forgot about that, thanks!